src/trading/TakeOrder.sol

```diff
@@ -0,0 +1,393 @@
+        });
+    }
+
+    function getTokensToCover(Direction _direction, uint256 _sharePriceRange, uint256 _sharePriceLong, uint256 _sharePriceShort, uint256 _numShares) internal constant returns (uint256) {
+        return _numShares
+            .mul((_direction == Direction.Long) ? _sharePriceLong : _sharePriceShort)
+            .div(_sharePriceRange)
+        ; // move semicolon up a line when https://github.com/duaraghav8/Solium/issues/110 is fixed
+    }
+
+    function getShortShareTokens(IMarket _market, uint8 _longOutcome) private constant returns (IShareToken[] memory) {
+        IShareToken[] memory _shortShareTokens = new IShareToken[](_market.getNumberOfOutcomes() - 1);
+        for (uint8 _outcome = 0; _outcome < _shortShareTokens.length + 1; ++_outcome) {
+            if (_outcome == _longOutcome) {
+                continue;
+            }
+            uint8 _index = (_outcome < _longOutcome) ? _outcome : _outcome - 1;
+            _shortShareTokens[_index] = _market.getShareToken(_outcome);
+        }
+        return _shortShareTokens;
+    }
+
+    function getSharePriceDetails(IMarket _market, IOrders _orders, bytes32 _orderId) private constant returns (uint256 _sharePriceRange, uint256 _sharePriceLong, uint256 _sharePriceShort) {
+        int256 _maxDisplayPrice = _market.getMaxDisplayPrice();
+        int256 _minDisplayPrice = _market.getMinDisplayPrice();
+        int256 _orderDisplayPrice = _orders.getPrice(_orderId);
+        _sharePriceRange = uint256(_maxDisplayPrice - _minDisplayPrice);
+        _sharePriceLong = uint256(_orderDisplayPrice - _minDisplayPrice);
+        _sharePriceShort = uint256(_maxDisplayPrice - _orderDisplayPrice);
+        return (_sharePriceRange, _sharePriceLong, _sharePriceShort);
+    }
+
+    function getDirections(IOrders _orders, bytes32 _orderId) private constant returns (Direction _makerDirection, Direction _takerDirection) {
+        return (_orders.getTradeType(_orderId) == Order.TradeTypes.Bid)
+            ? (Direction.Long, Direction.Short)
+            : (Direction.Short, Direction.Long)
+        ; // move semicolon up a line when https://github.com/duaraghav8/Solium/issues/110 is fixed
+    }
+
+    function getTakerSharesToSell(IShareToken _longShareToken, IShareToken[] memory _shortShareTokens, address _taker, Direction _takerDirection, uint256 _takerSize) private constant returns (uint256) {
+        uint256 _sharesAvailable = 2**256-1;
+        if (_takerDirection == Direction.Short) {
+            _sharesAvailable = _longShareToken.balanceOf(_taker);
+        } else {
+            for (uint8 _outcome = 0; _outcome < _shortShareTokens.length; ++_outcome) {
+                _sharesAvailable = _shortShareTokens[_outcome].balanceOf(_taker).min(_sharesAvailable);
+            }
+        }
+        return _sharesAvailable.min(_takerSize);
+    }
+}
+
+
+library DirectionExtensions {
+    function toTradeType(Trade.Direction _direction) internal constant returns (Order.TradeTypes) {
```

Comment:
Do we still need this since I added getOrderTradingTypeFromTakerDirection() to Order.sol?

Reply:
At some point we need to merge this file and your `Order.sol` file, which includes merging `Direction`.  However, I'm going to put that off for a future PR, probably after we decide what we are going to do with the trading contracts in general.